src/esrs/aggregate.rs

```diff
@@ -20,69 +20,72 @@ pub trait Identifier {
         Self: Sized;
 }
 
-#[async_trait]
-/// The Eraser trait is responsible for erasing an aggregate instance from history.
-pub trait Eraser<
-    Event: Serialize + DeserializeOwned + Send + Sync,
-    Error: From<sqlx::Error> + From<serde_json::Error> + Send + Sync,
->
-{
-    /// `delete` should either complete the aggregate instance, along with all its associated events, or fail.
-    /// If the deletion succeeds only partially, it _must_ return an error.
-    async fn delete(&self, aggregate_id: Uuid) -> Result<(), Error>;
-}
-
-/// The AggregateManager is responsible for loading an aggregate from the store, mapping commands to events, and
-/// persisting those events in the store. Be careful when implementing this trait, as you will be responsible for
-/// threading AggregateState/Commands/Events correctly.  For example, a bad implementation could result in an AggregateState
-/// that could not be replicated on load.
+/// The Aggregate trait is responsible for validating commands, mapping commands to events, and applying
+/// events onto the state.
 ///
-/// Unless you need to perform side effects as part of your command handling/verification you should implement the
-/// safer `Aggregate` trait instead.
-#[async_trait]
-pub trait AggregateManager: Identifier {
+/// An Aggregate should be able to derive its own state from nothing but its initial configuration, and its
+/// event stream. Applying the same events, in the same order, to the same aggregate, should always yield an
+/// identical aggregate state.
+///
+/// This trait is purposefully _synchronous_. If you are implementing this trait, your aggregate
+/// should not have any side effects. If you need additional information to handle commands correctly, then
+/// consider looking up that information and placing it in the command.
+pub trait Aggregate: Identifier {
```

Comment:
I'm not sure if Aggregate should extend Identifier. The latter is only used in the context of a store, while Aggregate is responsible for business logic, so the two are unrelated.